frontend/src/shared/util/hooks.ts

```diff
@@ -0,0 +1,36 @@
+/**
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import { useCallback, useEffect, useState } from 'react';
+
+export function useBackgroundRefresh (action: () => void, deps: readonly unknown[] = [], condition = true): boolean {
```

Comment:
Some nit picks. I'd love to see a docblock comment explaining this function, giving an example. The documentation I'm sure would explain it but because condition is such a generic name I am not really sure what I should be using that for. I'd maybe change it to `shouldRefresh` or `needsRefresh` maybe?